target/linux/ramips/mt7621/base-files/lib/preinit/03_mount_overlay

```diff
@@ -0,0 +1,57 @@
+
+wait_for_overlay_mtd_dev() {
+	while true; do
+		if [ -e "$(find_mtd_part 'overlay')" ]; then
+			break
+		fi
+		sleep 1
+	done
+}
+
+do_mount_overlay() {
+	[ ! -f /lib/upgrade/nand.sh ] && return
+
+	. /lib/functions.sh
+	. /lib/upgrade/nand.sh
+
+	local _root_dev=
+	local _data_dev=
+	local _data_idx=
+
+	_data_idx=$(find_mtd_index "overlay")
+
+	if [ -z "$_data_idx" ]; then
+		return
+	fi
+
+	wait_for_overlay_mtd_dev
+
+	ubiattach /dev/ubi_ctrl -m "$_data_idx"
+
+	# remove rootfs_data from root dev
+	_root_dev="/dev/$(nand_find_ubi 'ubi')"
+	if ubinfo "$_root_dev" -a 2>/dev/null|grep -qs rootfs_data; then
+		ubirmvol "$_root_dev" -N rootfs_data
+	fi
```

Comment:
I do not like this special overlay fs handling. 
If you want to use this extra space you can extend the ubi partition over multiple mtd partitions using `mtd-concat`, see https://github.com/openwrt/openwrt/blob/main/target/linux/ramips/dts/mt7621_sercomm_dxx_nand_256m.dtsi#L60 for example.